File: Back-end/NovaVerse/Models/Favorite.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace NovaVerse.Models
{
    public class Favorite
    {
        [Key, DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }

        public int UserId { get; set; }
        public virtual User User { get; set; }

        public int? ArtworkId { get; set; }
        public virtual Artwork Artwork { get; set; }

        public int? ArtistId { get; set; }
        public virtual User Artist { get; set; } 

        public DateTime CreateDate { get; set; }
    }
}

File: Front-end/NovaVerse/src/app/Models/favorite.ts
import { Artwork } from './artwork';

export interface Favorite {
  id: number;
  userId: number;
  artwork?: Artwork;  // Opera d'arte collegata al preferito
  artworkId?: number;
  artistId?: number | null;  // ID dell'artista se il preferito è un artista
  createDate: Date;
}

